opentaskspal/src/main/java/org/dmfs/opentaskspal/readdata/TaskUri.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.opentaskspal.readdata;
+
+import android.content.ContentUris;
+import android.net.Uri;
+
+import org.dmfs.android.contentpal.RowDataSnapshot;
+import org.dmfs.jems.single.Single;
+import org.dmfs.tasks.contract.TaskContract.Tasks;
+
+
+/**
+ * {@link Single} for the content {@link Uri} that refers to the given {@link RowDataSnapshot}.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class TaskUri implements Single<Uri>
```

Comment:
We definitely need to think of a better way to retrieve the `_ID` or `Uri` of a `RowReference`. There are other use cases in which this is necessary and we don't always have a `RowDataSnapshot` of the row, like when we just inserted it.
We'll keep this for now, but we need to fix that in `ContentPal`.